dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/dex/DexApiSyntax.scala

```diff
@@ -23,7 +24,23 @@ object DexApiSyntax {
       waitForOrderStatus(order.assetPair, order.id(), status)
 
     def waitForOrderStatus(assetPair: AssetPair, id: Order.Id, status: HttpOrderStatus.Status): F[HttpOrderStatus] =
-      waitForOrder(assetPair, id)(_.status == status)
+      waitForOrder(assetPair, id) { s =>
+
+        def fail() = throw new RuntimeException(s"Expected status $status for order $id, but found ${s.status}")
+        def check(): Boolean = s.status == status
+
+        status match {
+          case Filled | PartiallyFilled => s.status match {
+              case Cancelled | NotFound => fail()
```

Comment:
I think, we don't need `NotFound` here, because in `BouncingBalancesTestSuite` we have:
```scala
counterOrders.foreach(dex1.api.place)
submittedOrders.foreach(dex1.api.place)

counterOrders.foreach(dex1.api.waitForOrderStatus(_, Status.Filled))
submittedOrders.foreach(dex1.api.waitForOrderStatus(_, Status.Filled))
```

So, it is possible, that an order hasn't yet been placed and we fail checks.

Reply:
ok